packages/query-bar/src/components/QueryUrl.js

```diff
@@ -0,0 +1,163 @@
+import React, { useState, useRef } from 'react';
+import clsx from 'clsx';
+import {
+  Button,
+  IconButton,
+  Tooltip,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  withStyles,
+} from '@material-ui/core';
+import CopyIcon from '../assets/CopyIcon.svg';
+
+const QueryUrl = ({
+  classes,
+  filterItems,
+  localFind = {},
+  rootPath,
+}) => {
+  const [display, setDisplay] = useState(false);
+  const toggleDisplay = () => setDisplay(!display);
+
+  const [open, toggleOpen] = useState(false);
+
+  const pathFilterParams = filterItems.reduce((acc, item) => {
+    const { datafield, items = [] } = item;
+    acc[datafield] = items;
+    return acc;
+  }, {});
+
+  const query = JSON.stringify({
+    ...pathFilterParams,
+    ...localFind,
+  });
+  const url = encodeURI(rootPath.concat(query));
+
+  const copyUrl = async () => {
+    toggleOpen(!open);
```

Comment:
Same issue here, setting the state.

ex.
`toggleOpen((prevOpen) => !prevOpen);`